ppb/events.py

```diff
@@ -127,23 +129,23 @@ class StartScene:
         * `signal(new_scene=StartScene(MyScene(player=player))`
         * `signal(new_scene=StartScene, kwargs={"player": player}`
     """
-    new_scene: Union[Scene, Type[Scene]]
+    new_scene: Union[BaseScene, Type[BaseScene]]
```

Comment:
This is a confusing type signature.  Why is it either a subclass of `BaseScene` or an instance thereof?

Reply:
That's exactly what's going on.

The Engine either accepts a pre-constructed Scene instance, or it accepts a scene class and will construct it for you.

Reply:
OK, why is it `Type[BaseScene]` rather than `Callable[..., BaseScene]` then?  (i.e. shouldn't we allow any sort of callable there, not only class constructors?)

Reply:
I think it actually checks for a `type` to determine if it should call it.

Reply:
Thanks for all the explanations  <3
Changing that seems out-of-scope here, but should I make an issue for it?

Reply:
I'm not sure?

Basically, you need to determine the right shape to check for, and the obvious choices are:

* Checking for instances of `type` (disallowing non-class factories)
* Checking for callables (disallowing callable scenes)

Neither of these are super-great trade-offs? But also, neither of them are encouraged patterns.

Reply:
Fair.